gateway/gateway.go

```diff
@@ -1248,6 +1259,7 @@ func (gateway *HandleT) StartWebHandler(ctx context.Context) error {
 	srvMux.HandleFunc("/", WithContentType("application/json; charset=utf-8", app.LivenessHandler(gateway.jobsDB))).Methods("GET")
 	srvMux.HandleFunc("/v1/import", gateway.webImportHandler).Methods("POST")
 	srvMux.HandleFunc("/v1/audiencelist", gateway.webAudienceListHandler).Methods("POST")
+	srvMux.HandleFunc("/internal/v1/extract", gateway.webExtractHandler).Methods("POST")
```

Comment:
Thanks for the changes @debanjan97 ! Two more things.

1. Can we have the internals under a different mux?

```go
srvMux := mux.NewRouter()
internalMux := srvMux.PathPrefix("/internal").Subrouter()

internalMux.HandleFunc("/v1/extract", gateway.webExtractHandler).Methods("POST")
```

2. Can we have an HTTP middleware that checks if the request path begins with `/internal` and if it does return a 401 if a token in the `Authorization` header is not set? For now just checking if the token is not empty is OK, then in a later PR we can set an environment variable and compare against that.

Reply:
I was planning to add the middleware on the internal mux so that we can skip checking the request path.
That way, the middleware need not be restricted to the endpoints containing`/internal`; it can be used wherever we need authorization.

```go
internalMux := srvMux.PathPrefix("/internal").Subrouter() // mux for internal endpoints
internalMux.Use(
	middleware.InternalTokenMiddleware,
)
```

Reply:
In addition to that, currently, we overload the Authorization header for fetching the write key.
We have checks (not in any middleware, though) which reject requests if the `Authorization` header is missing.
https://github.com/rudderlabs/rudder-server/blob/b71ae92859b1dd9ac8720a0f54d7e05bad0ec26f/gateway/gateway.go#L943

Unless we decouple fetching the `writeKey` from the actual authorization, we might have to look for another way to reject calls to the internal endpoints without proper authorization.
Unless we are adding the middleware to currently act as a placeholder for what is next to come.

Reply:
@debanjan97 let's just add an internal mux then. we'll address the authorization in a later PR since it is more complex.

Reply:
addressed!